### ISSUE TITLE
Turning off session restore

### ISSUE THREAD

**Issue by fonnesbeck** [jupyterlab/jupyterlab#7330]
Each time I run JupyterLab it restores the session from the previous time running it, so all of the notebooks that I may have had running last time pop up again. Is there any way of turning this behavior off? Its a huge pain to have to close a bunch of old notebooks that I'm not using anymore, not to mention the resources they pull when they spin up automatically.

Running JupyerLab 1.1.4 on Linux and Python 3.6.7.

**Comment by ian-r-rose** [comment]
@fonnesbeck yep, this question was frequently-asked enough that it was added to the docs: https://jupyterlab.readthedocs.io/en/stable/getting_started/faq.html#tips-and-tricks (complete with a broken markdown link, apparently).

**Comment by jasongrout** [comment]
> complete with a broken markdown link, apparently).

Yeah, the docs are in rst, not markdown.

**Comment by fonnesbeck** [comment]
So, what's the flag? Its not clear to me from the `--help` output.

**Comment by jasongrout** [comment]
`--NotebookApp.default_url = '/lab?reset'`

**Comment by jasongrout** [comment]
(or you can add the setting to your config like shown in the docs so you don't have to add it to the command line each time)

**Comment by vidartf** [comment]
You should also be able to completely disable workspaces by this config: `LabApp.workspaces_dir=''`. This might cause the front-end to throw some errors if you try to use any of the workspace functionality though. (CC @afshin, using `?reset` when `workspaces_dir` is empty causes the app to splash screen to not disappear). Maybe this should be added to the docs?

**Comment by fonnesbeck** [comment]
On version 1.1.4 of JupyterLab, it claims not to recognize that setting:

`[C 15:13:20.230 LabApp] Invalid argument: '--NotebookApp.default_url'`

Tried to set it in the config json, but same result.

**Comment by ian-r-rose** [comment]
Can you show exactly the command line command you used? This just worked for me on 1.1.4
```bash
jupyter lab --NotebookApp.default_url=/lab?reset
```

Also, note that to put this in your config you should use a `jupyter_notebook_config.py`.

**Comment by vidartf** [comment]
Yeah try without the spaces as how @ian-r-rose wrote it.

**Comment by fonnesbeck** [comment]
I get the following:

```
$ jupyter lab --NotebookApp.default_url=/lab?reset
zsh: no matches found: --NotebookApp.default_url=/lab?reset
```

**Comment by vidartf** [comment]
Consulting `jupyter lab --help-all` locally, I see it listed as `LabApp.default_url`, you might try that.

**Comment by fonnesbeck** [comment]
Unfortunately, running: 

    jupyter lab --LabApp.default_url='/lab?reset'

does not suppress session restore (nor does `NotebookApp.default_url`.

**Comment by fonnesbeck** [comment]
Also, it looks like setting the option in my config file no longer works either (JupyterLab 1.2.3). It does take me to the URL above, but similarly fully restores my previous session.

**Comment by afshin** [comment]
> I get the following:
> 
> ```
> $ jupyter lab --NotebookApp.default_url=/lab?reset
> zsh: no matches found: --NotebookApp.default_url=/lab?reset
> ```
> 
> However, it does work when added to the config file, so I am all set. Thanks.

This is a zsh error, you needed to escape the `?`:
```
 jupyter lab --NotebookApp.default_url=/lab\?reset
```

However, I'm seeing an issue with the URL that is constructed so the reset parameter is not coming through (e.g.: `http://localhost:8888/lab?reset=&amp;token=XXXXXXXXXX`)

**Comment by fonnesbeck** [comment]
Correct, this does not reset the session for me.

**Comment by fonnesbeck** [comment]
Is this the best place to get the issue addressed, or should I create another issue?

Seems like it would be worth having a config flag for restore, rather than having to hack the URL. This is a pretty dramatic environment setting to not be able to permanently disable it.

**Comment by afshin** [comment]
I changed this to a `help wanted` issue. It can be accomplished purely in Python, so hopefully someone who might not be inclined to work on front-end issues might want to work on this one.

**Comment by jasongrout** [comment]
To make this more precise, what should the option be named? 

`jupyter lab --LabApp.clear_default_workspace=True`?

**Comment by fonnesbeck** [comment]
Something along the lines of `restore_workspace=False` might be clearer, perhaps? What appears to be happening is the state of the workspace is being saved on exit, similar to what happens on VSCode, so I'm trying to override this.

**Comment by afshin** [comment]
The way I imagine implementing this could be to check for that option and
have the REST endpoints in the `jupyterlab_server` simply not save incoming
workspaces.

**Comment by ihleonard-c3** [comment]
Adding c.NotebookApp.default_url = '/lab?reset' to my config doesn't seem to be working for me (jupyterlab 3.2.9) -- I still get put back in the same setup as my last session. This is particularly irritating for larger notebooks and cloud deployments

**Comment by krassowski** [comment]
Not sure if it helps, but please note that `NotebookApp` is legacy; you may be using a legacy file which may not be loaded or the setting may not get used; maybe [migrating to jupyter server](https://jupyter-server.readthedocs.io/en/stable/operators/migrate-from-nbserver.html) config could help?

**Comment by gbrault** [comment]
Working with version 3.3.4 (latest from the date of this post)

From this thread, it's not clear what is the good option to start Jupyter Lab while not restarting the previous session kernels.
Like most of the people in this thread:

```sh
$jupyter lab --NotebookApp.default_url=/lab\?reset or $jupyter lab --LabApp.default_url=/lab\?reset

does not work for me while starting on the root directory for jupyter lab: 
it stops immediatly telling the /lab directory is not found

$jupyter lab --LabApp.clear_default_workspace=True does not yel but does not work (kernel still restart)
```
The issue I am facing is that the server dies at boot, I need to restart lab once or twice (even more some time) to boot jupyter lab. This is an issue.

**Comment by AhmedMKhattab** [comment]
> Unfortunately, running:
> 
> ```
> jupyter lab --LabApp.default_url='/lab?reset'
> ```
> 
> does not suppress session restore (nor does `NotebookApp.default_url`. The page ends up being located at:
> 
> ```
> http://localhost:8888/lab?reset=&amp;token=<token_value>
> ```
> 
> yet, all the tabs I had opened last time are restored.
> 
> This seems like an insane default. If I had a large number of notebooks open in the previous session, they would all get spun up again. As an example, the session that was just restored for me immediately sucks up 11GB of memory. The alternative is to manually close each notebook tab before closing. Session restore should be something you deliberately turn on, not default behavior.

I'm having the same issue, anyone found a solution yet?

**Comment by halleysfifthinc** [comment]
I'm still encountering this issue and appending the "reset" query to the default_url does not work.

My current work-around is

```python
import tempfile
c.LabApp.workspaces_dir = tempfile.mkdtemp()
```
